### PR TITLE
feat (batcher): post batch to ethereum

### LINKED ISSUE #181: feat (batcher): post batch to ethereum
Post batch data to ethereal after posting to s3 bucket

### PR DESCRIPTION
**Pull request by taturosati**
Batcher now posts batches to ethereum. Reads configuration and deployment output from yaml files.

# To test:
1. Start anvil, aggregator, operator & batcher (make sure to have .env file filled with s3 credentials)
2. Run `make batcher_send_dummy_task` twice
3. Batcher should post 2 task batch to s3 and then to ethereum, then operator should verify both proofs & send response to aggregator

### CODE REVIEW

File: batcher/src/config/mod.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+pub struct ContractDeploymentOutput {
+    pub addresses: Addresses,
+}
+
+impl ContractDeploymentOutput {
+    pub fn new(deployment_output: String) -> Self {
+        let deployment_output = std::fs::read_to_string(deployment_output).expect("Failed to read deployment output file");
+        serde_json::from_str(&deployment_output).expect("Failed to parse deployment output file")
+    }
+}
```
Comment (JuArce) [inline batcher/src/config/mod.rs]: missing new line

File: batcher/src/eth/mod.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+pub async fn create_new_task(service_manager: AlignedLayerServiceManager, merkle_root: [u8; 32], batch_data_pointer: String) -> Result<TransactionReceipt, anyhow::Error> {
+    let call = service_manager.create_new_task(merkle_root, batch_data_pointer);
+    let pending_tx = call.send().await?;
+
+    match pending_tx.await? {
+        Some(receipt) => Ok(receipt),
+        None => Err(anyhow::anyhow!("Receipt not found")),
+    }
+}
```
Comment (JuArce) [inline batcher/src/eth/mod.rs]: missing new line

File: batcher/src/lib.rs [hunk 3]
```diff
@@ -197,6 +212,22 @@ impl App {
                 .expect("Failed to upload object to S3");
 
             info!("Batch sent to S3 with name: {}", file_name);
+
+            info!("Uploading batch to contract");
+            // generate random hash until we have merkle trees
```
Comment (JuArce) [inline batcher/src/lib.rs]: Add TODO keyword to find it easily

**Review by JuArce — CHANGES_REQUESTED** [review]
```
warning: unused import: `bytes::Bytes`
 --> src/s3/mod.rs:3:5
  |
3 | use bytes::Bytes;
  |     ^^^^^^^^^^^^
```
This import is unused but the files was not change in this pr

File: batcher/src/lib.rs (hunk 3 above)
Comment (taturosati) [reply batcher/src/lib.rs]: Added on 414828cc5c6f28b21ce825e501acd5db88372a73

File: batcher/src/eth/mod.rs (hunk 2 above)
Comment (taturosati) [reply batcher/src/eth/mod.rs]: Added on 414828cc5c6f28b21ce825e501acd5db88372a73

File: batcher/src/config/mod.rs (hunk 1 above)
Comment (taturosati) [reply batcher/src/config/mod.rs]: Added on 414828cc5c6f28b21ce825e501acd5db88372a73

File: batcher/src/config/mod.rs [hunk 4]
```diff
@@ -0,0 +1,40 @@
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+pub struct ECDSAConfig {
+    pub private_key_store_path: String,
+    pub private_key_store_password: String,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct BatcherConfigFromYaml {
```
Comment (entropidelic) [inline batcher/src/config/mod.rs:10]: maybe this could be named just `BatcherConfig`

**Review by entropidelic — APPROVED** [review]
lgtm 🚀

**Review by JuArce — APPROVED** [review]
lgtm

**Pull request merged**